src/components/contexts/FillLevelContext.tsx

```diff
@@ -0,0 +1,21 @@
+import { createContext, useContext } from 'react'
+
+export type FillLevel = 0 | 1 | 2 | 3
+
+export const FillLevelContext = createContext<FillLevel>(0)
+export const FillLevelProvider = FillLevelContext.Provider
+export const useFillLevel = () => useContext(FillLevelContext)
+
+export const increaseFillLevel: Record<FillLevel, FillLevel> = {
+  0: 1,
+  1: 2,
+  2: 3,
+  3: 3,
+}
```

Comment:
You can also use `Math.min(x + 1, 3)`.

Reply:
I was trying to battle the TS checker 🤦‍♂️ with that, but I came up with a better solution now. :) @maciaszczykm PTAL